controllers/drplacementcontrol_controller.go

```diff
@@ -1761,23 +1758,37 @@ func (r *DRPlacementControlReconciler) updateDRPCStatus(
 	return nil
 }
 
+// updateResourceCondition updates DRPC status sub-resource with updated status from VRG if one exists,
+// - The status update is NOT intended for a VRG that should be cleaned up on a peer cluster
+// It also updates DRPC ConditionProtected based on current state of VRG.
 func (r *DRPlacementControlReconciler) updateResourceCondition(
-	drpc *rmn.DRPlacementControl,
-	clusterName, vrgNamespace string,
-	log logr.Logger,
+	drpc *rmn.DRPlacementControl, userPlacement client.Object,
 ) {
-	annotations := make(map[string]string)
+	vrgNamespace, err := selectVRGNamespace(r.Client, r.Log, drpc, userPlacement)
+	if err != nil {
+		r.Log.Info("failed to select VRG namespace", "error", err)
+
+		return
+	}
+
+	clusterName := r.clusterForVRGStatus(drpc, userPlacement, r.Log)
```

Comment:
The old code is only called `updateResourceCondition` if and only if we have a valid cluster.   `clusterForVRGStatus` can return an empty string for the `clusterName`.  We should return in that case? And I think that's what you meant to do (from the log message)

Reply:
Yes, if clusterForVRGStatus returns an empty cluster name then skip the resource conditions update.